### PR TITLE
Dont include data elems when rendering relationship sections.  Our AP…

### PR DESCRIPTION
**Pull request by mattcollum**
…I populates these dependent on what the service does so better not to return them ever vs sporadically

### CODE REVIEW

**Comment by mattcollum** [comment]
Looks like I checked in a version of go.mod in our api project that points to this branch (and it works) so seems safe to merge.  Figured you should take a look regardless though

File: node.go [hunk 1]
```diff
@@ -68,14 +68,30 @@ type Node struct {
...
 // RelationshipManyNode is used to represent a generic has many JSON API
 // relation
 type RelationshipManyNode struct {
+	Data  []*Node `json:"-"`
+	Links *Links  `json:"links,omitempty"`
+	Meta  *Meta   `json:"meta,omitempty"`
+}
+
+// RelationshipOneNodeInput is used to represent a generic has one JSON API relation
+// for marshaling without data elem
```
Comment (ntayefeh) [inline node.go]: with data element?

Comment (ntayefeh) [reply node.go]: ah nvm, I see it's only use for unmarshalling to support cases that data element doens't exist, but the marshalls are all forced to ignore data using the non-input structs

Comment (mattcollum) [reply node.go]: Good catch.  I originally was going to use the new struct for the output scenario but switched it around when it was less complex to go the other way

**Comment by mattcollum** [comment]
NOT merging for now as the admin apis use this

**Pull request closed without merging**